src/EditorFeatures/Core.Wpf/InlineRename/Dashboard/Dashboard.xaml.cs

```diff
@@ -316,8 +317,19 @@ private void Apply_Click(object sender, RoutedEventArgs e)
 
         private void Commit()
         {
-            _model.Session.Commit();
-            _textView.VisualElement.Focus();
+            try
+            {
+                _model.Session.Commit();
+                _textView.VisualElement.Focus();
+            }
+            catch(NotSupportedException ex)
+            {
+                // Session.Commit can throw if it can't commit
+                // rename operation.
+                // handle that case gracefully
+                var notificationService = _model.Session.Workspace.Services.GetService<INotificationService>();
+                notificationService.SendNotification(ex.Message, title: EditorFeaturesResources.Rename, severity: NotificationSeverity.Error);
```

Comment:
1. this feels a bit wonky to use exceptions for what sounds like an expected scenario (i.e. not being able to commit).    Can we make Commit just return a message instead of throwing?
2. This seems to wrap too many lines.  It feels lke this should just wrap the .Commit call, not the .Focus call.